### DISCUSSION TITLE
Acknowledgements and question

### DISCUSSION THREAD

**Discussion by zacatecnik** [weltenwort/home-assistant-rct-power-integration#257] [Q&A]
Hi, first of all, I want to thank you. Great integration. I'm sorry, I'm a beginner and a layman. I want to ask you for advice. Do you know why and how to fix that all my entities are not showing(settings->control panels->energy) ? Specifically if I want to fill in "add consumption" I don't see for example the entities sensor.rct_power_storage_consumer_power, sensor.rct_power_storage_grid_power

Thank you very much

**Comment by weltenwort** [comment]
Hi @zacatecnik :wave:

The entities you mention show the instantaneous *power* used in that moment. But the built-in Home Assistant energy dashboard requires *energy* entities (with units like Wh) that are the integral of the power over time. It filters the drop-down in the dashboard settings to only show entities with the correct physical quantities. This integration also provides these:

- for energy consumed from the grid: `sensor.rct_power_grid_energy_consumption_total`
- for energy returned to the grid: `sensor.rct_power_grid_energy_production_absolute_total` (the "absolute" in the name is important to make sure the sign is as HA expects it)

These and the other entities to track solar production and battery levels on the dashboard are also explained in the [readme section "Usage with the built-in energy dashboard"](https://github.com/weltenwort/home-assistant-rct-power-integration#usage-with-the-built-in-energy-dashboard).

Does that make sense? Let me know if this helps. :four_leaf_clover:

**Reply by zacatecnik** [reply]
Hey. Hey,

Thank you very much for your quick and helpful reply. Oh, I thought HA only needs to see instantaneous values to display correctly. So are you saying that I should only select values with the absolute total property in the items ?
Meant to add consumption, return, solar generation, batt system 
Do I understand this correctly ? 

Edit: and did you have a problem with sensor.rct_power_storage_grid_energy_production_total with a negative value ? I I have -1395251.2

**Reply by weltenwort** [reply]
Indeed, the energy dashboard uses the energy values for all those.

> did you have a problem with `sensor.rct_power_storage_grid_energy_production_total`

Yes, the inverter returns a value with a negative sign. That's why I also added `sensor.rct_power_grid_energy_production_absolute_total` to take the absolute value. You should be able to use that for the dashboard.

**Comment by zacatecnik** [comment]
Hey, thanks. I think I got it :-)
But I'm surprised that it shows me quite different values compared to the original application. 

Thanks again for your help and helpfulness.

**Reply by weltenwort** [reply]
Glad to hear it's working now.

> [...] But I'm surprised that it shows me quite different values compared to the original application.

The integration reads the values using the same protocol as the app. But the way that Home Assistant calculates them might be different. Which values specifically are different for you?
